mlflow/server/js/src/setupAjaxHeaders.js

```diff
@@ -0,0 +1,32 @@
+import $ from 'jquery';
+import cookie from 'cookie';
+
+// To enable running behind applications that require specific headers
+// to be set during HTTP requests (e.g., CSRF tokens), we support parsing
+// a set of cookies with a key prefix of "mlflow-request-header-$HeaderName",
+// which will be added as an HTTP header to all AJAX requests.
+export const setupAjaxHeaders = () => {
+  const requestHeaders = getRequestHeaders();
+  $.ajaxSetup({
```

Comment:
I am not really familiar with js, but I looked up docs [ajax setup]((https://api.jquery.com/jquery.ajaxsetup/) ) and it says 

> Note: Global callback functions should be set with their respective global Ajax event handler methods—.ajaxStart(), .ajaxStop(), .ajaxComplete(), .ajaxError(), .ajaxSuccess(), .ajaxSend()—rather than within the options object for $.ajaxSetup().

Did we consider that?

Reply:
Also, it looks like we assume the set of cookies for the site will not change during the session?